src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -392,14 +569,32 @@ UCS_CLASS_INIT_FUNC(uct_tcp_sockcm_ep_t, const uct_ep_params_t *params)
 ucs_status_t uct_tcp_sockcm_ep_create(const uct_ep_params_t *params, uct_ep_h *ep_p)
 {
     uct_tcp_sockcm_ep_t *tcp_ep;
+    uct_tcp_sockcm_t *tcp_sockcm;
+    ucs_status_t status;
 
     if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR) {
         /* create a new endpoint for the client side */
         return UCS_CLASS_NEW(uct_tcp_sockcm_ep_t, ep_p, params);
     } else if (params->field_mask & UCT_EP_PARAM_FIELD_CONN_REQUEST) {
+        tcp_ep     = (uct_tcp_sockcm_ep_t*)(params->conn_request);
+        tcp_sockcm = uct_tcp_sockcm_ep_get_cm(tcp_ep);
+        UCS_ASYNC_BLOCK(tcp_sockcm->super.iface.worker->async);
```

Comment:
why need lock here?

Reply:
this function can be called from both threads, tcp_ep's state can be changed from both.

Reply:
ok